### DISCUSSION TITLE
Can TEKSI's wastewater (QGEP) and drinking water (QWAT) modules be officially tested with PostgreSQL 16?

### DISCUSSION THREAD

**Discussion by nicolas-sitylb** [teksi/Home#103] [Q&A]
Hello,

Can TEKSI's wastewater (QGEP) and drinking water (QWAT) modules be officially tested with PostgreSQL 16 ([first released in September 2023](https://www.postgresql.org/about/news/postgresql-16-released-2715/))?
This will enable us to benefit from the latest bug fixes and improvements constantly being made by the PostgreSQL team.

\>>> https://github.com/teksi/Home/wiki/TEKSI-modules-releases-and-roadmap <<<

Also what is the policy behind that roadmap?
Is there a strict and regular schedule for testing PostgreSQL/PostGIS/QGIS versions, for example using an automated process (CD/CI)?

Thank you.

**Comment by ponceta** [comment]
Hi @nicolas-sitylb,
It's a very good question!
As we proposed yet, and as example for 2024 we would test :
future versions (postgres 16, QGIS 3.36, ...)
New LTR / current versions ( postgres 15, QGIS 3.34, ...)
Previous LTR ( postgres 14, QGIS 3.28)
And in the past we used to assess the oldest version supported but this was not a good signal for the users to remain on obsolete versions of postgres like postgreSQL 9.
We also figured out that on testing with CI using docker images, some images like postgis are not well updated so testing can show false positive results.
